jujugui/static/gui/src/app/subapps/browser/browser.js

```diff
@@ -175,30 +175,26 @@ YUI.add('subapp-browser', function(Y) {
       // Hold onto charm data so we can pass model instances to other views when
       // charms are selected.
       this._cache = new Y.juju.BrowserCache();
-      this.state = new models.UIState({
-        // Disallow routing to inspectors if we are in sandbox mode; the
-        // model to be inspected will not be available.
-        allowInspector: !cfg.sandbox,
-        baseUrl: cfg.baseUrl || '',
-        dispatchers: {
-          app: {
-            deployTarget: this._deployTargetDispatcher.bind(this)
-          },
-          sectionA: {
-            charmbrowser: this._charmBrowserDispatcher.bind(this),
-            inspector: this._inspectorDispatcher.bind(this),
-            empty: this.emptySectionA.bind(this),
-            services: this._addedServicesDispatcher.bind(this)
-          },
-          sectionB: {
-            machine: this._machine.bind(this),
-            empty: this.emptySectionB.bind(this)
-          }
-        }
-      });
 
       this._registerSubappHelpers();
 
+      this.state = cfg.state
+      this.state.set('dispatchers', {
```

Comment:
Maybe we could set this up as a merge now so we don't have to in the future. At the least maybe a comment that this needs to be updated to merge.